File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        char[] aux = word.toCharArray();
        int start = 0, last = 0;
        while (last < word.length() && word.charAt(last) != ch) last++;
        if (last == word.length()) return word;
        while (start < last) {
            char temp = aux[start];
            aux[start] = aux[last];
            aux[last] = temp;
            start++;
            last--;
        }
        return new String(aux);
    }
}

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.cpp
class Solution {
public:
    void reverse(string &word, int start, int end) {
        while (start < end) {
            int t = word[start];
            word[start] = word[end];
            word[end] = t;
            start++;
            end--;
        }
    }
    
    string reversePrefix(string word, char ch) {
        int N = word.size();
        
        int i = 0;
        
        while (i < N) {
            if (word[i] == ch) {
                break;
            }
            i++;
        }
        
        if (i != N) {
            reverse(word, 0, i);
        }
        
        return word;
    }
};